### PR TITLE
Adding CDN commands with PnP PowerShell

### PR DESCRIPTION
**Pull request by KoenZomers**
Article on Office 365 CDN contained commands for SPO Management Shell and O365 CLI, but not yet for PnP PowerShell. Added them to this article as an option.

### CODE REVIEW

**Comment by mypil** [comment]
@JohanFreelancer9 - kindly do the copy editor review for this PR. Thanks.

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 1]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
 
 </details>
 
+<a name="CDNSetupinPnPPosh"> </a>
+## Set up and configure the Office 365 CDN by using PnP PowerShell
+
+The procedures in this section require you to use PnP PowerShell to connect to SharePoint Online. For instructions, see [Getting started with PnP PowerShell](https://github.com/SharePoint/PnP-PowerShell#getting-started).
+
+Complete these steps to set up and configure the CDN to host your assets in SharePoint Online using PnP Powershell.
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
Complete these steps to set up and configure the CDN to host your assets in SharePoint Online using PnP PowerShell.
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 2]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+
+<details>
+  <summary>Click to expand</summary>
+
+### Enable your organization to use the Office 365 CDN
+
+Before you make changes to the tenant CDN settings, you should retrieve the current status of the private CDN configuration in your Office 365 tenant. Connect to your tenant using PnP Powershell:
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
Before you make changes to the tenant CDN settings, you should retrieve the current status of the private CDN configuration in your Office 365 tenant. Connect to your tenant using PnP PowerShell:
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 3]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+
+Now use the **Get-PnPTenantCdnEnabled** cmdlet to retrieve the CDN status settings from the tenant:
+
+``` powershell
+Get-PnPTenantCdnEnabled -CdnType <Public | Private>
+```
+
+The status of the CDN for the specified CdnType will output to the screen.
+
+Use the **Set-PnPTenantCdnEnabled** cmdlet to enable your organization to use the Office 365 CDN. You can enable your organization to use public origins, private origins, or both at once. You can also configure the CDN to skip the setup of default origins when you enable it. You can always add these origins later as described in this topic.
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
Use the **Set-PnPTenantCdnEnabled** cmdlet to enable your organization to use the Office 365 CDN. You can enable your organization to use public origins, private origins, or both at at the same time. You can also configure the CDN to skip the setup of default origins when you enable it. You can always add these origins later as described in this topic.
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 4]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+  
+In PnP Powershell:
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
In PnP PowerShell:
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 5]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+Set-PnPTenantCdnEnabled -CdnType Public -Enable $true
+```
+
+To enable your organization to use private origins, type the following command:
+
+``` powershell
+Set-PnPTenantCdnEnabled -CdnType Private -Enable $true
+```
+
+For more information about this cmdlet, see [Set-PnPTenantCdnEnabled](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/set-pnptenantcdnenabled).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about this cmdlet, see [Set-PnPTenantCdnEnabled](https://docs.microsoft.com/powershell/module/sharepoint-pnp/set-pnptenantcdnenabled).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 6]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+Set-PnPTenantCdnPolicy -CdnType Private -PolicyType IncludeFileExtensions -PolicyValue "CSS,PNG"
+```
+
+To see what file types are currently allowed by the CDN, use the **Get-PnPTenantCdnPolicies** cmdlet:
+
+``` powershell
+Get-PnPTenantCdnPolicies -CdnType <Public | Private>
+```
+
+For more information about these cmdlets, see [Set-PnPTenantCdnPolicy](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/set-pnptenantcdnpolicy) and [Get-PnPTenantCdnPolicies](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/get-pnptenantcdnpolicies).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about these cmdlets, see [Set-PnPTenantCdnPolicy](https://docs.microsoft.com/powershell/module/sharepoint-pnp/set-pnptenantcdnpolicy) and [Get-PnPTenantCdnPolicies](https://docs.microsoft.com/powershell/module/sharepoint-pnp/get-pnptenantcdnpolicies).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 7]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+The _IncludeFileExtensions_ property contains the list of file extensions that will be served from the CDN.
+
+> [!NOTE]
+> The default file extensions are different between public and private.
+
+The _ExcludeRestrictedSiteClassifications_ property contains the site classifications that you want to exclude from the CDN. For example, you can exclude sites marked as **Confidential** so content from sites with that classification applied will not be served from the CDN.
+
+The _ExcludeIfNoScriptDisabled_ property excludes content from the CDN based on the site-level _NoScript_ attribute settings. By default, the _NoScript_ attribute is set to **Enabled** for _Modern_ sites and **Disabled** for _Classic_ sites. This depends on your tenant settings.
+
+For more information about these cmdlets, see [Set-PnPTenantCdnPolicy](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/set-pnptenantcdnpolicy) and [Get-PnPTenantCdnPolicies](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/get-pnptenantcdnpolicies).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about these cmdlets, see [Set-PnPTenantCdnPolicy](https://docs.microsoft.com/powershell/module/sharepoint-pnp/set-pnptenantcdnpolicy) and [Get-PnPTenantCdnPolicies](https://docs.microsoft.com/powershell/module/sharepoint-pnp/get-pnptenantcdnpolicies).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 8]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+``` powershell
+Add-PnPTenantCdnOrigin -CdnType Public -OriginUrl */masterpage
+```
+
++ The wildcard modifier ***/** can only be used at the beginning of the path, and will match all URL segments under the specified URL.
++ The path can point to a document library, folder or site. For example, the path _*/site1_ will match all the document libraries under the site.
+
+You can add an origin with a specific relative path. You cannot add an origin using the full path.
+
+This example adds a private origin of the siteassets library on a specific site:
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
This example adds a private origin of the site assets library on a specific site:
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 9]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+
+``` powershell
+Add-PnPTenantCdnOrigin -CdnType Private -OriginUrl sites/test/siteassets/folder%201
+```
+
+``` powershell
+Add-PnPTenantCdnOrigin -CdnType Private -OriginUrl "sites/test/siteassets/folder 1"
+```
+
+For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 10]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+  Add-PnPTenantCdnOrigin -CdnType Public -OriginUrl */style%20library
+  ```
+
++ Use the **Add-PnPTenantCdnOrigin** cmdlet to define the master pages as a public origin.
+
+``` powershell
+  Add-PnPTenantCdnOrigin -CdnType Public -OriginUrl */masterpage
+  ```
+
+For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 11]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+  Add-PnPTenantCdnOrigin -CdnType Private -OriginUrl */sitepages
+  ```
+
++ Use the **Add-PnPTenantCdnOrigin** cmdlet to define the publishing images folder as a private origin.
+
+``` powershell
+  Add-PnPTenantCdnOrigin -CdnType Private -OriginUrl */publishingimages
+  ```
+
+For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 12]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+<a name="ExamplePrivateOriginSiteCollectionPnPPosh"> </a>
+### Example: Configure a private origin for a site collection for SharePoint Online
+
+Use the **Add-PnPTenantCdnOrigin** cmdlet to define a site collection as a private origin. For example:
+
+``` powershell
+Add-PnPTenantCdnOrigin -CdnType Private -OriginUrl sites/site1/siteassets
+```
+
+For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about this command and its syntax, see [Add-PnPTenantCdnOrigin](https://docs.microsoft.com/powershell/module/sharepoint-pnp/add-pnptenantcdnorigin).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 13]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+### Manage the Office 365 CDN
+
+Once you've set up the CDN, you can make changes to your configuration as you update content or as your needs change, as described in this section.
+  
+<a name="Office365CDNforSPOaddremoveassetPnPPosh"> </a>
+#### Add, update, or remove assets from the Office 365 CDN
+
+Once you've completed the setup steps, you can add new assets, and update or remove existing assets whenever you want. Just make your changes to the assets in the folder or SharePoint library that you identified as an origin. If you add a new asset, it is available through the CDN immediately. However, if you update the asset, it will take up to 15 minutes for the new copy to propagate and become available in the CDN.
+  
+If you need to retrieve the location of the origin, you can use the **Get-PnPTenantCdnOrigin** cmdlet. For information on how to use this cmdlet, see [Get-PnPTenantCdnOrigin](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/get-pnptenantcdnorigin).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
If you need to retrieve the location of the origin, you can use the **Get-PnPTenantCdnOrigin** cmdlet. For information on how to use this cmdlet, see [Get-PnPTenantCdnOrigin](https://docs.microsoft.com/powershell/module/sharepoint-pnp/get-pnptenantcdnorigin).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 14]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+<a name="Office365CDNforSPORemoveOriginPnPPosh"> </a>
+#### Remove an origin from the Office 365 CDN
+
+You can remove access to a folder or SharePoint library that you identified as an origin. To do this, use the **Remove-PnPTenantCdnOrigin** cmdlet.
+
+``` powershell
+Remove-PnPTenantCdnOrigin -OriginUrl <path> -CdnType <Public | Private | Both>
+```
+
+For information on how to use this cmdlet, see [Remove-PnPTenantCdnOrigin](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/remove-pnptenantcdnorigin).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For information on how to use this cmdlet, see [Remove-PnPTenantCdnOrigin](https://docs.microsoft.com/powershell/module/sharepoint-pnp/remove-pnptenantcdnorigin).
```

File: Enterprise/use-office-365-cdn-with-spo.md [hunk 15]
```diff
@@ -438,6 +439,289 @@ For more information about this cmdlet, see [Set-SPOTenantCdnEnabled](https://te
...
+Set-PnPTenantCdnEnabled -CdnType Public -Enable $false
+```
+
+To disable use of the private origins in the CDN, enter the following command:
+
+``` powershell
+Set-PnPTenantCdnEnabled -CdnType Private -Enable $false
+```
+
+For more information about this cmdlet, see [Set-PnPTenantCdnEnabled](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/set-pnptenantcdnenabled).
```
Comment (JohanFreelancer9) [inline Enterprise/use-office-365-cdn-with-spo.md]:
```suggestion
For more information about this cmdlet, see [Set-PnPTenantCdnEnabled](https://docs.microsoft.com/powershell/module/sharepoint-pnp/set-pnptenantcdnenabled).
```

**Comment by KoenZomers** [comment]
@JohanFreelancer9, thanks for your crisp view on this. I've committed all your suggested changes.
@mypil, it's ready to be merged.

**Comment by mypil** [comment]
@kelleyvice-msft - PR has been copy edited and is ready for final review. 

Thank you.

cc: @kenwith

**Comment by kenwith** [comment]
Merging per @kelleyvice-msft

**Pull request merged**